platform/src/core/utils/useFetchAnalyticsData.jsx

```
@@ -0,0 +1,97 @@
+  const [allSiteData, setAllSiteData] = useState([]);
+  const [chartLoading, setChartLoading] = useState(true); // Start loading only initially
+  const [error, setError] = useState(null);
+  const refreshChart = useSelector((state) => state.chart.refreshChart);
+
+  const getValidDateRange = useCallback(() => {
+    let { startDate, endDate } = dateRange;
+
+    if (!(startDate instanceof Date)) startDate = new Date(startDate);
+    if (!(endDate instanceof Date)) endDate = new Date(endDate);
+
+    if (isNaN(startDate) || isNaN(endDate) || startDate > endDate) {
```

Biome

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)